async-nats/src/options.rs

```diff
@@ -160,6 +161,91 @@ impl ConnectOptions {
         }
     }
 
+    /// Authenticate with a JWT. Requires function to sign the server nonce.
+    /// The signing function is synchronous and should not do any blocking io.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    /// let seed = "SUANQDPB2RUOE4ETUA26CNX7FUKE5ZZKFCQIIW63OX225F2CO7UEXTM7ZY";
+    /// let kp = nkeys::KeyPair::from_seed(seed).unwrap();
+    /// // load jwt from creds file or other secure source
+    /// async fn load_jwt() -> std::io::Result<String> { todo!(); }
+    /// let jwt = load_jwt().await?;
+    ///
+    /// let nc = async_nats::ConnectOptions::with_jwt(jwt, move |nonce| kp.sign(nonce).unwrap())
+    ///     .connect("localhost").await?;
+    /// # std::io::Result::Ok(())
+    /// # }
+    /// ```
+    pub fn with_jwt<S>(jwt: String, sig_cb: S) -> Self
+    where
+        S: Fn(&[u8]) -> Vec<u8> + Send + Sync + 'static,
+    {
+        ConnectOptions {
+            auth: Authorization::Jwt(
+                jwt,
+                Arc::new(move |nonce| Ok(base64_url::encode(&sig_cb(nonce)))),
+            ),
+            ..Default::default()
+        }
+    }
+
+    /// Authenticate with NATS using a `.creds` file.
+    /// Open the provided file, load its creds,
+    /// and perform the desired authentication
+    ///
+    /// # Example
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    /// let nc = async_nats::ConnectOptions::with_credentials("path/to/my.creds".into()).await?
+    ///     .connect("connect.ngs.global").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn with_credentials(path: PathBuf) -> io::Result<Self> {
```

Comment:
Naming nit between this and `with_static_credentials`. The lowest form (e.g a primitive in memory type) seems like the natural fit to be named the simpler name (e.g `with_credentials`).

```suggestion
    pub async fn with_credentials_file(path: PathBuf) -> io::Result<Self> {
```

Reply:
done